src/core/maptoscreen.cpp

```diff
@@ -77,3 +78,36 @@ void MapToScreen::transformPoint()
   }
   emit screenPointChanged();
 }
+
+void MapToScreen::setMapDistance( const double distance )
+{
+  if ( mMapDistance == distance )
+    return;
+
+  mMapDistance = distance;
+  emit mapDistanceChanged();
+  transformDistance();
+}
+
+double MapToScreen::mapDistance() const
+{
+  return mMapDistance;
+}
+
+double MapToScreen::screenDistance() const
+{
+  return mScreenDistance;
+}
+
+void MapToScreen::transformDistance()
+{
+  if ( !mMapSettings || qgsDoubleNear( mMapDistance, 0.0 ) )
+  {
+    mScreenDistance = 0.0;
+  }
+  else
+  {
+    mScreenDistance = mMapDistance / mMapSettings->mapUnitsPerPoint();
```

Comment:
Can we add a safe guard in case `mMapSettings->mapUnitsPerPoint()` returns zero?